analyses/cell-type-ewings/scripts/aucell-ews-signatures/01-aucell.R

```diff
@@ -0,0 +1,192 @@
+    help = "Path to directory containing custom gene sets to use with AUCell. 
+      All TSV files in the directory will be used and must contain the `ensembl_gene_id` column.
+      File names will be used as the name of the gene set."
+  ),
+  make_option(
+    opt_str = c("--max_rank_threshold"),
+    type = "double",
+    default = 0.01,
+    help = "Percentage of genes detected to set as the `aucMaxRank`. 
+      Must be a number between 0 and 1."
+  ),
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to file where results will be saved"
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2025,
+    help = "A random seed for reproducibility."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure input files exist
+stopifnot(
+  "sce file does not exist" = file.exists(opt$sce_file),
+  "max_rank_threshold must be between 0 and 1" = (opt$max_rank_threshold <= 1 & opt$max_rank_threshold > 0)
+)
+
+# load SCE
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+})
+
+
+# set up multiprocessing params
+if (opt$threads > 1) {
+  bp_param <- BiocParallel::MulticoreParam(opt$threads)
+} else {
+  bp_param <- BiocParallel::SerialParam()
+}
+
+# check that max rank is number between 0-1
+
+# make sure directory exists for writing output
+output_dir <- dir(opt$output_file)
+fs::dir_create(output_dir)
+
+# read in SCE
+sce <- readr::read_rds(opt$sce_file)
+
+# remove genes that are not detected from SCE object 
+genes_to_remove <- rowData(sce)$detected > 0 
+filtered_sce <- sce[genes_to_remove , ]
+
+# Prep gene sets ---------------------------------------------------------------
+
+# get list of files with custom gene sets to use 
+gene_set_files <- list.files(opt$custom_geneset_dir, recursive = TRUE, full.names = TRUE)
+# get names of gene sets using name of the files 
+gene_set_names <- stringr::str_replace(basename(gene_set_files), ".tsv", "")
+
+# read in custom gene sets
+custom_genes_list <- gene_set_files |> 
+  purrr::set_names(gene_set_names) |> 
+  purrr::map(\(file) {
+    gene_ids <- readr::read_tsv(file) |> 
+      dplyr::pull(ensembl_gene_id) |>
+      unique()
+  })
+
+# grab msigdb gene sets 
+ews_gene_sets <- c(
+  "staege" = "STAEGE_EWING_FAMILY_TUMOR",
+  "miyagawa_up" = "MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_UP",
+  "miyagawa_down" = "MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN",
+  "zhang"= "ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION", 
+  "riggi_up" = "RIGGI_EWING_SARCOMA_PROGENITOR_UP",
+  "riggi_down" = "RIGGI_EWING_SARCOMA_PROGENITOR_DN",
+  "kinsey_up" = "KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_UP",
+  "kinsey_down"= "KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_DN"
+)
```

Comment:
I'm actually wondering now whether we want to make this a stand-alone TSV file to read in here, and we can format it into a named vector in the script since I think that may be more maintainable should anything change here.

I'm also wondering about the Wrenn msigdb sets as seen in `references/README.md`.. Do we want to use those here?
 
```
- [`HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.html)
- [`REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/GOBP_REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION.html)
```